src/common/perf_counters.cc

```diff
@@ -190,7 +190,15 @@ void PerfCounters::dec(int idx, uint64_t amt)
   ceph_assert(!(data.type & PERFCOUNTER_LONGRUNAVG));
   if (!(data.type & PERFCOUNTER_U64))
     return;
-  data.u64 -= amt;
+  if (amt > data.u64) { // would wrap
+    assert(data.u64 >= amt); // Assert in debug builds
```

Comment:
seems we cannot enable this `assert()` yet
```
2020-10-30T04:53:22.460 INFO:teuthology.orchestra.run.smithi117:> sudo ceph config set mgr mgr/devicehealth/enable_monitoring false --force
2020-10-30T04:53:22.677 INFO:teuthology.orchestra.run.smithi117.stderr:python3.6: /build/ceph-16.0.0-6749-g874c7de6/src/common/perf_counters.cc:194: void ceph::common::PerfCounters::dec(int, uint64_t): Assertion `data.u64 >= amt' failed.
2020-10-30T04:53:22.679 DEBUG:teuthology.orchestra.run:got remote process result: None
2020-10-30T04:53:22.679 ERROR:teuthology.contextutil:Saw exception from nested tasks
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/contextutil.py", line 31, in nested
    vars.append(enter())
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2020-10-29-2343/qa/tasks/ceph.py", line 1412, in run_daemon
    remote.run(args=cmd.split(' '))
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/remote.py", line 215, in run
    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 446, in run
    r.wait()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 160, in wait
    self._raise_for_status()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 178, in _raise_for_status
    raise CommandCrashedError(command=self.command)
teuthology.exceptions.CommandCrashedError: Command crashed: 'sudo ceph config set mgr mgr/devicehealth/enable_monitoring false --force'
```

see https://pulpito.ceph.com/kchai-2020-10-30_02:27:32-rados-wip-kefu-testing-2020-10-29-2343-distro-basic-smithi/5572986/


and https://pulpito.ceph.com/kchai-2020-10-30_02:27:32-rados-wip-kefu-testing-2020-10-29-2343-distro-basic-smithi/

Reply:
OK, but I'm just going to comment it out as we should be able to have this assert at some time in the future. It is an unsigned type after all.